src/test/java/org/ibp/api/java/impl/middleware/dataset/validator/ObservationsTableValidatorTest.java

```diff
@@ -116,32 +118,81 @@ public void testValidateObservationsValuesDataTypesMissingValue() throws Excepti
 	}
 
 	@Test(expected = ApiRequestValidationException.class)
-	public void testValidateObservationsValuesDataTypesInvalidDate() throws Exception {
+	public void testValidateObservationsValuesDataTypesInvalidDate() {
 		final Table<String, String, String> data = HashBasedTable.create();
 		data.put("Obs1", "A", "A");
-		MeasurementVariable measurementVariable = new MeasurementVariable();
+		final MeasurementVariable measurementVariable = new MeasurementVariable();
 		measurementVariable.setAlias("A");
 		measurementVariable.setDataTypeId(TermId.DATE_VARIABLE.getId());
 		final List<MeasurementVariable> measurementVariables = new ArrayList<>();
 		measurementVariables.add(measurementVariable);
 		try {
 			observationsTableValidator.validateObservationsValuesDataTypes(data, measurementVariables);
-		} catch (ApiRequestValidationException e) {
+		} catch (final ApiRequestValidationException e) {
 			assertThat(Arrays.asList(e.getErrors().get(0).getCodes()), hasItem("warning.import.save.invalidCellValue"));
 			throw e;
 		}
 	}
 
 	@Test
-	public void testValidateObservationsValuesDataTypesValidDate() throws Exception {
+	public void testValidateObservationsValuesDataTypesValidDate() {
 		final Table<String, String, String> data = HashBasedTable.create();
 		data.put("Obs1", "A", "20181010");
-		MeasurementVariable measurementVariable = new MeasurementVariable();
+		final MeasurementVariable measurementVariable = new MeasurementVariable();
 		measurementVariable.setAlias("A");
 		measurementVariable.setDataTypeId(TermId.DATE_VARIABLE.getId());
 		final List<MeasurementVariable> measurementVariables = new ArrayList<>();
 		measurementVariables.add(measurementVariable);
 		observationsTableValidator.validateObservationsValuesDataTypes(data, measurementVariables);
 	}
 
+	@Test(expected = ApiRequestValidationException.class)
+	public void testValidateObservationsValuesDataTypesNoCategoricalDefinedWithImportValue() {
+		final Table<String, String, String> data = HashBasedTable.create();
+		data.put("Obs1", "A", "1");
+		final MeasurementVariable measurementVariable = new MeasurementVariable();
+		measurementVariable.setAlias("A");
+		measurementVariable.setDataTypeId(TermId.CATEGORICAL_VARIABLE.getId());
+		final List<MeasurementVariable> measurementVariables = new ArrayList<>();
+		measurementVariables.add(measurementVariable);
+		try {
+			observationsTableValidator.validateObservationsValuesDataTypes(data, measurementVariables);
+		} catch (final ApiRequestValidationException e) {
+			assertThat(Arrays.asList(e.getErrors().get(0).getCodes()), hasItem("warning.import.save.invalidCategoricalValue"));
+			throw e;
+		}
+	}
+
+	@Test(expected = ApiRequestValidationException.class)
+	public void testValidateObservationsValuesDataTypesNoCategoricalDefinedNoImportValue() {
```

Comment:
rename to testValidateObservationsValuesDataTypesNoCategoricalScaleDefinedNoImportValue.

Reply:
Good test case